### ISSUE TITLE
trying to install friends encoder getting cors error

### ISSUE THREAD

**Issue by JoshWho** [WWBN/AVideo-Encoder#435]
for some reason it is getting a cors error when it get to the install part.

https://encoder.corder.tv/install/index.php

when entering the info and everything click install it just hangs and inspect element show a cors error with https://corder.tv/login something like that.

Is there something I can add to his main site to say this domain is allowed? and where would that go if so?

![error](https://user-images.githubusercontent.com/39392449/169784337-691620f4-37ec-401e-a114-30dbf16283a0.png)

**Comment by DanielnetoDotCom** [comment]
It is probably not a clean ubuntu installation and something is blocking access/requests from different domains.

may be some cors rules somewhere on the server installation

**Comment by JoshWho** [comment]
It is a fresh install kinda I was going off old directions and it started to install another avideo. I deleted the other avideo should I just wipe it and do it again?

**Comment by DanielnetoDotCom** [comment]
I mean, the streamer site is blocking. it is something on his apache.

maybe a security module

**Comment by JoshWho** [comment]
Is there a cors file or is this just something htaccess does? I do not remember doing anything special with mine

**Comment by JoshWho** [comment]
I wonder if the ssl sets a cors

**Comment by DanielnetoDotCom** [comment]
by default avideo leave all cors rules opened.

it is something specific on your web server installation (yes maybe the ssl)

**Comment by JoshWho** [comment]
Is there a way to run the install without it needing to login to the main site?

**Comment by DanielnetoDotCom** [comment]
Yes but the problem will still persists 

you must resolve the cors problem , otherwise the encoder will be useless

**Comment by JoshWho** [comment]
only thing I can think is the ssl. Must be something letsencrypt changed.

**Comment by JoshWho** [comment]
Access to XMLHttpRequest at 'https://corder.tv/login' from origin 'https://encoder.corder.tv' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header contains multiple values 'https://encoder.corder.tv, *', but only one is allowed.

**Comment by JoshWho** [comment]
looks like a typo in the cors I think

**Comment by JoshWho** [comment]
This is the ip http://162.210.100.119/ I do not understand this but will figure it out eventually

**Comment by DanielnetoDotCom** [comment]
Could you install it using the IP?

**Comment by JoshWho** [comment]
So even just using the ip it still does the cors thing. Is there a different encoder for just a Standalone Encoder server. The only instructions I found shows how to install this with avideo.

https://www.linuxbabe.com/ubuntu/install-avideo-youphptube-ubuntu-20-04-server

**Comment by DanielnetoDotCom** [comment]
Remove any cord rule from Apache.conf

also make sure you did install Apache header module

**Comment by JoshWho** [comment]
> Remove any cord rule from Apache.conf
> 
> also make sure you did install Apache header module

what u mean by cord rule?

**Comment by DanielnetoDotCom** [comment]
I mean cors, it was a typo

**Comment by JoshWho** [comment]
> I mean cors, it was a typo

I do not see anything that has cors to it in the confs

**Comment by JoshWho** [comment]
I think i found it it is the ssl   includeSubDomains;  was missing will test and write back

**Comment by JoshWho** [comment]
I think it was the hsts on the domain messing with it. I have new dns settings taking effect fingers crossed will check back later.

**Comment by JoshWho** [comment]
OK it might not be that at all either. It just won't let me install it no matter what I change

**Comment by JoshWho** [comment]
Is it because there is no video uploaded yet to the main site that it is just redirecting and not truly doing the cors maybe?

**Comment by DanielnetoDotCom** [comment]
is very hard for me to guess, that is why I am asking you to remove all access-control rules from the apache.conf and other confs.

usually the AVideo access-control is enough

**Comment by JoshWho** [comment]
http://162.210.100.119/

https://corder.tv/

https://encoder.corder.tv/install/index.php


Still wondering if this will change if I upload somthing

**Comment by JoshWho** [comment]
Any clue what access control could be hiding that I am not seeing. I looked in every conf and even in the htaccess and still do not see anywhere why the script is loading: access-control-allow-origin: https://imasdk.googleapis.com/, *

It is only the website reporting that. When I change the doc directory to a empty folder that Any clue why it shows access-control-allow-origin: https://imasdk.googleapis.com/, *  is no longer there so I know it is something Avideo is doing

**Comment by DanielnetoDotCom** [comment]
try to install the encoder using another streamer site.

then check if it still works and if you can log in to the original site.

if you are using Cloudflare, disable the Cloudflare proxy.

**Comment by JoshWho** [comment]
It lets me try my site and the test site. When I try a brand new install it shows that  access-control-allow-origin: https://imasdk.googleapis.com/, * in the header

**Comment by JoshWho** [comment]
Go here and you can see the error  https://cors-test.codehappy.dev/?url=https%3A%2F%2Fcorder.tv%2F&method=get


This is a fresh Ubuntu 20 and Fresh install of Avideo

**Comment by JoshWho** [comment]
> try to install the encoder using another streamer site.
> 
> then check if it still works and if you can log in to the original site.
> 
> if you are using Cloudflare, disable the Cloudflare proxy.

No cloudflare it is all direct dns with the host. that is what is stumping me about this. Cloudflare I can just force the header to be whatever I want but I do not have that setup on there yet.

**Comment by DanielnetoDotCom** [comment]
I found this piece of code

https://github.com/WWBN/AVideo/blob/897b69cf12b8d4d5d7eb3d03a067a38f45adc518/objects/functions.php#L3213-L3225

why is your $_SERVER['HTTP_ORIGIN'] empty?

by default, you should send it.

we detect if it is empty and we need this line so we can allow video ads on the website

**Comment by DanielnetoDotCom** [comment]
I just sent an update to the AVideo, check if makes any difference for you

**Comment by JoshWho** [comment]
Still doing the same thing. Just git pulled and cleared the cache

**Comment by JoshWho** [comment]
> I found this piece of code
> 
> https://github.com/WWBN/AVideo/blob/897b69cf12b8d4d5d7eb3d03a067a38f45adc518/objects/functions.php#L3213-L3225
> 
> why is your $_SERVER['HTTP_ORIGIN'] empty?
> 
> by default, you should send it.
> 
> we detect if it is empty and we need this line so we can allow video ads on the website

Where do I find this to replace what is missing?

**Comment by JoshWho** [comment]
Chrome

**Comment by JoshWho** [comment]
I do not get that error when I change the doc root to a empty folder.

**Comment by DanielnetoDotCom** [comment]
did you try to install this encoder using your joshwho site as the default site?

**Comment by JoshWho** [comment]
yes it works with my site

**Comment by DanielnetoDotCom** [comment]
so clearly the issue is on your friend's server installation.

**Comment by JoshWho** [comment]
yes but it is only on the avideo script. When I change the doc directory to a folder with a single index file i do not have that error.

**Comment by DanielnetoDotCom** [comment]
correct, AVideo requires the $_SERVER['HTTP_REFERER'] or $_SERVER['HTTP_ORIGIN'] , otherwise It cannot work properly

in your case both are blank

**Comment by JoshWho** [comment]
this encoder https://encoder1.wwbn.net/?justLogin=1&PHPSESSID=bsghm073n85p89kgdkelfick66 works with the sote https://corder.tv so I do not think that is the issue

**Comment by JoshWho** [comment]
> correct, AVideo requires the $_SERVER['HTTP_REFERER'] or $_SERVER['HTTP_ORIGIN'] , otherwise It cannot work properly
> 
> in your case both are blank

I can't find anything online about how to correct this on the server. Everything says that is a browser issue.

**Comment by DanielnetoDotCom** [comment]
is very hard for me to guide you on it too.

did you install only the defaults apache modules?

there is no proxy on it?

**Comment by JoshWho** [comment]
correct. I just did the step by step tutorial nothing extra

**Comment by DanielnetoDotCom** [comment]
Why don't you try this

https://github.com/WWBN/AVideo/wiki/How-to-install-LAMP,-FFMPEG-and-Git-on-a-fresh-Ubuntu-20.x-for-AVideo-Platform-version-11.x-or-newer

and just delete the AVideo folder when finishes

**Comment by JoshWho** [comment]
I keep getting alomost done and this i get this error 

ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

**Comment by JoshWho** [comment]
Im gonna just wait for him to get back to me So I can just have a re install a fresh install again. I deleted everything did the purge and for some reason still has a myqsl connection on reboot

There should be a option in linux to remove service esily

**Comment by JoshWho** [comment]
OK got the vps back to a fresh install just ran the install you gave me and Now I am back to the cors error again.

Trying to add encoder.corder.tv to corder.tv still gives a cors error

**Comment by JoshWho** [comment]
Could you please give a look at this @DanielnetoDotCom I wish I was able to follow the instructions to a t and not need your help for this but it isn't working like it should.

**Comment by JoshWho** [comment]
Can't we just make like a sytatic header that just says to cors rules make it overwrite whatever it is seeing?

**Comment by DanielnetoDotCom** [comment]
> ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

This mean your mysql server is gone

**Comment by DanielnetoDotCom** [comment]
I do not have this error anywhere but try this ...

```
cd /var/www/html/AVideo-Encoder/install/
php install.php https://corder.tv/
```

this installation thru the command line does not check any cors

**Comment by JoshWho** [comment]
That worked perfectly Thank you @DanielnetoDotCom

**Comment by DanielnetoDotCom** [comment]
As I suspect. even if you install it, the error will persist due the cors problem

the problem is on the streamer, NOT on the encoder. I am not sure how you can deal with it

**Comment by JoshWho** [comment]
Yup I can't figure out where on corder.tv is the issue. I do not see anything wrong with the php or apache all looks the same as mine

**Comment by DanielnetoDotCom** [comment]
maybe the problem is on the DNS manager side

**Comment by JoshWho** [comment]
DNS is not using cloudflare that is just that tester using that. DNS is direct ip

**Comment by Broly5551** [comment]
Hello there, @DanielnetoDotCom !  This is Justin Derby, the founder of Corder and the person that Josh has been helping out in this thread.  In addition to the money spent on servers from JoshWho Hosting Solutions in order to host Corder with the AVideo script, I also spent roughly 290 dollars buying a bunch of the plugins off your site in order to make Corder as feature-rich as possible.

The CORS issue is making it impossible for me to use the encoder on my site, and this issue is the only thing preventing me from rolling Corder out to the world.  If you could take some time to sit down and look at my server set-up, you may be able to figure what went wrong.  We can provide you with the server log-in credentials.

Please and thanks,

Justin

**Comment by DanielnetoDotCom** [comment]
Hi @Broly5551 

I also do not know where is the problem.

If I were in your shoes, I would hire me to make a clean installation for you and let me handle since the DNS configuration till the apache/mysql/avideo installation

I can not offer partial installations because it is a lot easier to make a clean install than find bugs on preinstalled systems.

Basic installation costs 70 USD, but we require a clean server for that.  To make a clean installation we require the following:

**Operational system**
You will need to provide a clean Ubuntu 20.04 or 22.04 (VPS or dedicated), shared servers do not work. (do not install a non LTS Ubuntu version)
Please make sure you do NOT install any kind of control panel (No Plesk, cPanel, etc)

**Hardware requirements**
About the server requirements please check this https://github.com/WWBN/AVideo/wiki/AVideo-Platform-Hardware-Requirements

**Comment by DanielnetoDotCom** [comment]
good news is that you have some progress.

but I am not sure how this "Encoder.php on line 1048" error went to the repository.

I am sending an update to fix it in a few minutes

**Comment by JoshWho** [comment]
Sweet thank you

**Comment by DanielnetoDotCom** [comment]
please let me know if works now

**Comment by JoshWho** [comment]
Video never makes it to the streamer site

![84ea3c00-c9d2-4441-9de1-22bd70d01f29](https://user-images.githubusercontent.com/39392449/172436916-fff96363-1d5a-4960-b08e-86f7e3447e5c.png)

**Comment by DanielnetoDotCom** [comment]
delete all the submitted videos, also delete all your encoder logs, submit a new video and share the whole log (feel free to hide the password hash)

**Comment by JoshWho** [comment]
```
[07-Jun-2022 17:43:42 Europe/London] logoff:: done session_id = 96ee2pntvmarkq8cmre62rnm6l
[07-Jun-2022 17:43:44 Europe/London] https://corder.tv/ == https://corder.tv/
[07-Jun-2022 17:43:44 Europe/London] login.json: Login::run
[07-Jun-2022 17:43:44 Europe/London] Login::run (admin, ***, https://corder.tv/, true)
[07-Jun-2022 17:43:44 Europe/London] Login::run request login user (admin)
[07-Jun-2022 17:43:44 Europe/London] Login::run request login complete user (admin)
[07-Jun-2022 17:43:44 Europe/London] Login::run got an object
[07-Jun-2022 17:43:44 Europe/London] createIfNotExists:: admin
[07-Jun-2022 17:43:44 Europe/London] Verification Start https://corder.tv/
[07-Jun-2022 17:43:44 Europe/London] Verification GetFrom Cache https://corder.tv/
[07-Jun-2022 17:43:44 Europe/London] Verification Response (): {"msg":["https:\/\/corder.tv\/ Site was inactive"],"isPrivate":false,"verified":true,"status":"i"}
[07-Jun-2022 17:43:44 Europe/London] Login:: almost done
[07-Jun-2022 17:43:44 Europe/London] Login:: done session_id = 96ee2pntvmarkq8cmre62rnm6l session_login 
[07-Jun-2022 17:43:58 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 17:43:58 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"96ee2pntvmarkq8cmre62rnm6l"}
[07-Jun-2022 17:43:58 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 17:43:58 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 17:43:58 Europe/London] PHP Fatal error:  Uncaught mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AND `resolution` = '' AND `format`= 'mp4' LIMIT 1' at line 1 in /var/www/html/AVideo-Encoder/objects/Upload.php:25
Stack trace:
#0 /var/www/html/AVideo-Encoder/objects/Upload.php(25): mysqli->query()
#1 /var/www/html/AVideo-Encoder/objects/Encoder.php(1048): Upload::loadFromEncoder()
#2 /var/www/html/AVideo-Encoder/view/upload.php(110): Encoder::sendFile()
#3 /var/www/html/AVideo-Encoder/view/jquery-file-upload/server/php/fileuploadchunkdone.php(30): require_once('...')
#4 {main}
  thrown in /var/www/html/AVideo-Encoder/objects/Upload.php on line 25

```

**Comment by DanielnetoDotCom** [comment]
I just send something to try to help debug it. can you please update and try again?

**Comment by JoshWho** [comment]
This time I seen it go from the uploader to the cue but error
![dc2d8ff7-6e16-4b7b-82de-b9e56ab8147c](https://user-images.githubusercontent.com/39392449/172438967-13a11adb-3db7-42cb-93a4-d0cba8f80f34.png)
```

[07-Jun-2022 17:51:44 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 17:51:44 Europe/London] decideFormatOrder: {"file":"video_210513160622_3001_HD.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"96ee2pntvmarkq8cmre62rnm6l"}
[07-Jun-2022 17:51:44 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 17:51:44 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 17:51:44 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:51:44 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:51:47 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] downloadFile: start queue_id = 1  url = https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037 pathFileName = /var/www/html/AVideo-Encoder/videos/1_tmpFile.mp4
[07-Jun-2022 17:51:47 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 17:51:47 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 17:51:47 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037 | url = https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037
[07-Jun-2022 17:51:47 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037
[07-Jun-2022 17:51:47 Europe/London] downloadFile: {"error":false,"filename":"1_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/1_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 17:51:47 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] {}

```

**Comment by JoshWho** [comment]
Is there something I can run that will re install the database?

**Comment by JoshWho** [comment]
k I will try that/


This is the log from the last test

```

[07-Jun-2022 17:51:44 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 17:51:44 Europe/London] decideFormatOrder: {"file":"video_210513160622_3001_HD.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"96ee2pntvmarkq8cmre62rnm6l"}
[07-Jun-2022 17:51:44 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 17:51:44 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 17:51:44 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:51:44 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:51:47 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] downloadFile: start queue_id = 1  url = https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037 pathFileName = /var/www/html/AVideo-Encoder/videos/1_tmpFile.mp4
[07-Jun-2022 17:51:47 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 17:51:47 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 17:51:47 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037 | url = https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037
[07-Jun-2022 17:51:47 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_video2105131606223001HD_YPTuniqid_629f8220ace0c6.12037037
[07-Jun-2022 17:51:47 Europe/London] downloadFile: {"error":false,"filename":"1_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/1_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 17:51:47 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:51:47 Europe/London] {}
[07-Jun-2022 17:57:57 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 17:57:57 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"96ee2pntvmarkq8cmre62rnm6l"}
[07-Jun-2022 17:57:57 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 17:57:57 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 17:57:57 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:57:57 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 17:57:58 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:57:58 Europe/London] downloadFile: start queue_id = 2  url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f83952dc445.53367415 pathFileName = /var/www/html/AVideo-Encoder/videos/2_tmpFile.mp4
[07-Jun-2022 17:57:58 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 17:57:58 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 17:57:58 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_BBC_YPTuniqid_629f83952dc445.53367415 | url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f83952dc445.53367415
[07-Jun-2022 17:57:58 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f83952dc445.53367415
[07-Jun-2022 17:57:58 Europe/London] downloadFile: {"error":false,"filename":"2_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/2_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 17:57:58 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:57:58 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 17:57:58 Europe/London] {}

```

**Comment by JoshWho** [comment]
```
[07-Jun-2022 18:04:01 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 18:04:01 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"96ee2pntvmarkq8cmre62rnm6l"}
[07-Jun-2022 18:04:01 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 18:04:01 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 18:04:01 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:04:01 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:04:06 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:04:06 Europe/London] downloadFile: start queue_id = 3  url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f850113ca22.16411831 pathFileName = /var/www/html/AVideo-Encoder/videos/3_tmpFile.mp4
[07-Jun-2022 18:04:06 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 18:04:06 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 18:04:06 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_BBC_YPTuniqid_629f850113ca22.16411831 | url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f850113ca22.16411831
[07-Jun-2022 18:04:06 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f850113ca22.16411831
[07-Jun-2022 18:04:06 Europe/London] downloadFile: {"error":false,"filename":"3_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/3_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 18:04:06 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:04:06 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:04:06 Europe/London] {}

```

**Comment by DanielnetoDotCom** [comment]
really weird, I cannot see the verification method on your log , you need to log in again. also clear your cookies just to make sure

also check the streamer side, if there is any error

**Comment by DanielnetoDotCom** [comment]
no need to drop tables, I just need you to login again and clear the cookies

**Comment by JoshWho** [comment]
ok fresh login cleaned cookies  

```
[07-Jun-2022 18:22:47 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 18:22:47 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"ude196hu2nnstn1441c3dcvagv"}
[07-Jun-2022 18:22:47 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 18:22:47 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 18:22:47 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:22:47 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:22:50 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:22:50 Europe/London] downloadFile: start queue_id = 4  url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f89671dcd97.78875916 pathFileName = /var/www/html/AVideo-Encoder/videos/4_tmpFile.mp4
[07-Jun-2022 18:22:50 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 18:22:50 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 18:22:50 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_BBC_YPTuniqid_629f89671dcd97.78875916 | url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f89671dcd97.78875916
[07-Jun-2022 18:22:50 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f89671dcd97.78875916
[07-Jun-2022 18:22:50 Europe/London] downloadFile: {"error":false,"filename":"4_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/4_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 18:22:50 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:22:50 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:22:50 Europe/London] {}

```

**Comment by DanielnetoDotCom** [comment]
the login logs are not shared on your last post, please share the whole log

**Comment by JoshWho** [comment]
that is a clean log I deleted everything before using it

**Comment by DanielnetoDotCom** [comment]
MUST be something regarding your login.

or you did not logoff or clear the cookies

**Comment by JoshWho** [comment]
I did I cleaned cookies and thats what it does.

I see this error in the que log try [1] Error on return_vars->videos_id

```
[07-Jun-2022 18:29:35 Europe/London] logoff:: done session_id = ude196hu2nnstn1441c3dcvagv
[07-Jun-2022 18:29:39 Europe/London] https://corder.tv/ == https://corder.tv/
[07-Jun-2022 18:29:39 Europe/London] login.json: Login::run
[07-Jun-2022 18:29:39 Europe/London] Login::run (admin, ***, https://corder.tv/, false)
[07-Jun-2022 18:29:39 Europe/London] Login::run request login user (admin)
[07-Jun-2022 18:29:39 Europe/London] Login::run request login complete user (admin)
[07-Jun-2022 18:29:39 Europe/London] Login::run got an object
[07-Jun-2022 18:29:39 Europe/London] createIfNotExists:: admin
[07-Jun-2022 18:29:39 Europe/London] https://corder.tv/ == https://corder.tv/
[07-Jun-2022 18:29:39 Europe/London] Verification Start https://corder.tv/
[07-Jun-2022 18:29:39 Europe/London] Verification GetFrom Cache https://corder.tv/
[07-Jun-2022 18:29:39 Europe/London] Verification Response (): {"msg":["https:\/\/corder.tv\/ Site was inactive"],"isPrivate":false,"verified":true,"status":"i"}
[07-Jun-2022 18:29:39 Europe/London] Login:: almost done
[07-Jun-2022 18:29:39 Europe/London] Login:: done session_id = ude196hu2nnstn1441c3dcvagv session_login 
[07-Jun-2022 18:29:57 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 18:29:57 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"ude196hu2nnstn1441c3dcvagv"}
[07-Jun-2022 18:29:57 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 18:29:57 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 18:29:57 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:29:57 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:30:03 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:30:03 Europe/London] downloadFile: start queue_id = 5  url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8b15f0e139.97333218 pathFileName = /var/www/html/AVideo-Encoder/videos/5_tmpFile.mp4
[07-Jun-2022 18:30:03 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 18:30:03 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 18:30:03 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_BBC_YPTuniqid_629f8b15f0e139.97333218 | url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8b15f0e139.97333218
[07-Jun-2022 18:30:03 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8b15f0e139.97333218
[07-Jun-2022 18:30:03 Europe/London] downloadFile: {"error":false,"filename":"5_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/5_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 18:30:03 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:30:03 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:30:03 Europe/London] {}

```

**Comment by DanielnetoDotCom** [comment]
NOW YOU SHARE IT

I suspect the error now is on your streamer site, but here is more code to debug.

I need you to logoff and login again

**Comment by JoshWho** [comment]
```
[07-Jun-2022 18:44:20 Europe/London] logoff:: done session_id = ude196hu2nnstn1441c3dcvagv
[07-Jun-2022 18:44:20 Europe/London] https://corder.tv/ == https://corder.tv/
[07-Jun-2022 18:44:20 Europe/London] login.json: Login::run
[07-Jun-2022 18:44:20 Europe/London] Login::run (admin, ***, https://corder.tv/, true)
[07-Jun-2022 18:44:20 Europe/London] Login::run request login user (admin)
[07-Jun-2022 18:44:21 Europe/London] Login::run request login complete user (admin)
[07-Jun-2022 18:44:21 Europe/London] Login::run got an object
[07-Jun-2022 18:44:21 Europe/London] createIfNotExists:: admin
[07-Jun-2022 18:44:21 Europe/London] Verification Start https://corder.tv/
[07-Jun-2022 18:44:21 Europe/London] Verification GetFrom Cache https://corder.tv/
[07-Jun-2022 18:44:21 Europe/London] Verification Response (): {"msg":["https:\/\/corder.tv\/ Site verified"],"isPrivate":false,"verified":true,"status":"a"}
[07-Jun-2022 18:44:21 Europe/London] Login:: almost done
[07-Jun-2022 18:44:21 Europe/London] Login:: done session_id = ude196hu2nnstn1441c3dcvagv session_login 
[07-Jun-2022 18:44:24 Europe/London] logoff:: done session_id = ude196hu2nnstn1441c3dcvagv
[07-Jun-2022 18:44:27 Europe/London] https://corder.tv/ == https://corder.tv/
[07-Jun-2022 18:44:27 Europe/London] login.json: Login::run
[07-Jun-2022 18:44:27 Europe/London] Login::run (admin, ***, https://corder.tv/, false)
[07-Jun-2022 18:44:27 Europe/London] Login::run request login user (admin)
[07-Jun-2022 18:44:28 Europe/London] Login::run request login complete user (admin)
[07-Jun-2022 18:44:28 Europe/London] Login::run got an object
[07-Jun-2022 18:44:28 Europe/London] createIfNotExists:: admin
[07-Jun-2022 18:44:28 Europe/London] Verification Start https://corder.tv/
[07-Jun-2022 18:44:28 Europe/London] Verification GetFrom Cache https://corder.tv/
[07-Jun-2022 18:44:28 Europe/London] Verification Response (): {"msg":["https:\/\/corder.tv\/ Site verified"],"isPrivate":false,"verified":true,"status":"a"}
[07-Jun-2022 18:44:28 Europe/London] Login:: almost done
[07-Jun-2022 18:44:28 Europe/London] Login:: done session_id = ude196hu2nnstn1441c3dcvagv session_login 
[07-Jun-2022 18:44:49 Europe/London] Upload.php will let function decide decideFormatOrder
[07-Jun-2022 18:44:49 Europe/London] decideFormatOrder: {"file":"BBC.mp4","audioOnly":"false","spectrum":"false","webm":"false","override_status":"","update_video_id":0,"inputHLS":"false","inputLow":"false","inputSD":"false","inputHD":"false","inputAutoHLS":"false","inputAutoMP4":"true","inputAutoWebm":"false","inputAutoAudio":"false","title":"","description":"","categories_id":0,"PHPSESSID":"ude196hu2nnstn1441c3dcvagv"}
[07-Jun-2022 18:44:49 Europe/London] decideFormatOrder: auto MP4
[07-Jun-2022 18:44:49 Europe/London] AVideo-Encoder Format::getFromOrder(7)
[07-Jun-2022 18:44:49 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:44:49 Europe/London] Encoder::sendFile encoder_id is empty {}
[07-Jun-2022 18:44:51 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:44:51 Europe/London] downloadFile: start queue_id = 9  url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8e911696a2.78194793 pathFileName = /var/www/html/AVideo-Encoder/videos/9_tmpFile.mp4
[07-Jun-2022 18:44:51 Europe/London] downloadFile: not using getYoutubeDl
[07-Jun-2022 18:44:51 Europe/London] downloadFile: this file was uploaded from file and thus is in the videos
[07-Jun-2022 18:44:51 Europe/London] downloadFile: downloadedFile = /var/www/html/AVideo-Encoder/videos/original_BBC_YPTuniqid_629f8e911696a2.78194793 | url = https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8e911696a2.78194793
[07-Jun-2022 18:44:51 Europe/London] Try to get UTF8 decode URL https://encoder.corder.tv/videos/original_BBC_YPTuniqid_629f8e911696a2.78194793
[07-Jun-2022 18:44:51 Europe/London] downloadFile: {"error":false,"filename":"9_tmpFile.mp4","pathFileName":"\/var\/www\/html\/AVideo-Encoder\/videos\/9_tmpFile.mp4","msg":"We downloaded the file with success"}
[07-Jun-2022 18:44:51 Europe/London] try [1] return_vars->videos_id is empty {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:44:51 Europe/London] getReturn_vars {"videos_id":0,"video_id_hash":""}
[07-Jun-2022 18:44:51 Europe/London] {}

```

**Comment by JoshWho** [comment]
I just used your encoder and it uploaded the video fine

**Comment by DanielnetoDotCom** [comment]
I am lost about what is happening. 

please make sure your encoder server can see the streamer server.

**Comment by JoshWho** [comment]
Should I just reinstall it? Drop the tables?

**Comment by JoshWho** [comment]
> I am lost about what is happening.
> 
> please make sure your encoder server can see the streamer server.

Yes able to ping it and also able to wget the website from the encoder server

**Comment by JoshWho** [comment]
Could you give me the command to drop the tables I am use to the gui

**Comment by DanielnetoDotCom** [comment]
should be something like this `drop database avideo-Encoder`

**Comment by DanielnetoDotCom** [comment]
does it work?

**Comment by JoshWho** [comment]
I have no clue what it did. The encoder server now shows the main website somehow like it is parked on top of it. No clue how it being 2 different servers. I am gonna do another fresh install tonight will let u know.

**Comment by DanielnetoDotCom** [comment]
ok, thanks
